[2024-06-14T06:16:15.552+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-14T06:16:15.564+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Spark_dag.query_job manual__2024-06-14T06:16:14.583392+00:00 [queued]>
[2024-06-14T06:16:15.573+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Spark_dag.query_job manual__2024-06-14T06:16:14.583392+00:00 [queued]>
[2024-06-14T06:16:15.577+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-06-14T06:16:15.589+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): query_job> on 2024-06-14 06:16:14.583392+00:00
[2024-06-14T06:16:15.593+0000] {standard_task_runner.py:63} INFO - Started process 231 to run task
[2024-06-14T06:16:15.595+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Spark_dag', 'query_job', 'manual__2024-06-14T06:16:14.583392+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/Spark_dag.py', '--cfg-path', '/tmp/tmp3gii5m27']
[2024-06-14T06:16:15.598+0000] {standard_task_runner.py:91} INFO - Job 4: Subtask query_job
[2024-06-14T06:16:15.649+0000] {task_command.py:426} INFO - Running <TaskInstance: Spark_dag.query_job manual__2024-06-14T06:16:14.583392+00:00 [running]> on host c099674506c1
[2024-06-14T06:16:15.709+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='Spark_dag' AIRFLOW_CTX_TASK_ID='query_job' AIRFLOW_CTX_EXECUTION_DATE='2024-06-14T06:16:14.583392+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-14T06:16:14.583392+00:00'
[2024-06-14T06:16:15.711+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-14T06:16:15.734+0000] {spark_submit.py:271} INFO - Could not load connection string Spark_connection, defaulting to yarn
[2024-06-14T06:16:15.736+0000] {spark_submit.py:401} INFO - Spark-Submit cmd: spark-submit --master yarn --conf spark.driver.memory=1g --conf spark.executor.memory=1g --conf spark.executor.instances=1 --name arrow-spark jobs/python/pyspark_test.py
[2024-06-14T06:16:15.804+0000] {spark_submit.py:571} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-06-14T06:16:15.806+0000] {spark_submit.py:571} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/spark-class: line 71: /usr/lib/jvm/java-11-openjdk-arm64/bin/java: No such file or directory
[2024-06-14T06:16:15.808+0000] {spark_submit.py:571} INFO - /home/***/.local/lib/python3.11/site-packages/pyspark/bin/spark-class: line 97: CMD: bad array subscript
[2024-06-14T06:16:15.809+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-14T06:16:15.817+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.11/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 502, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --conf spark.driver.memory=1g --conf spark.executor.memory=1g --conf spark.executor.instances=1 --name arrow-spark jobs/python/pyspark_test.py. Error code is: 1.
[2024-06-14T06:16:15.825+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=Spark_dag, task_id=query_job, run_id=manual__2024-06-14T06:16:14.583392+00:00, execution_date=20240614T061614, start_date=20240614T061615, end_date=20240614T061615
[2024-06-14T06:16:15.836+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 4 for task query_job (Cannot execute: spark-submit --master yarn --conf spark.driver.memory=1g --conf spark.executor.memory=1g --conf spark.executor.instances=1 --name arrow-spark jobs/python/pyspark_test.py. Error code is: 1.; 231)
[2024-06-14T06:16:15.861+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-14T06:16:15.875+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-14T06:16:15.877+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
